ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx

```diff
@@ -38,39 +71,536 @@ export const ClauseType: ClauseTypeMap = {
 
 interface FeatureAutoOpsRulesFormProps {
   featureId: string;
-  onSubmit: () => void;
+  refetchAutoOpsRules: () => void;
 }
 
 export const FeatureAutoOpsRulesForm: FC<FeatureAutoOpsRulesFormProps> = memo(
-  ({ featureId, onSubmit }) => {
-    const editable = useIsEditable();
-    const { formatMessage: f } = useIntl();
+  ({ featureId, refetchAutoOpsRules }) => {
+    const { operationId } = useParams<{ operationId: string }>();
+    const isNew = operationId === ID_NEW;
+    const dispatch = useDispatch<AppDispatch>();
+
+    const [selectedAutoOpsRule, setSelectedAutoOpsRule] =
+      useState<AutoOpsRule.AsObject | null>(null);
+
+    const autoOpsRules = useSelector<AppState, AutoOpsRule.AsObject[]>(
+      (state) =>
+        selectAllAutoOpsRules(state.autoOpsRules).filter(
+          (rule) => rule.featureId === featureId
+        ),
+      shallowEqual
+    );
+
+    const [open, setOpen] = useState(isNew);
+    const [isKillSwitchSelected, setIsKillSwitchSelected] = useState(false);
+    const history = useHistory();
+    const currentEnvironment = useCurrentEnvironment();
+
     const methods = useFormContext();
-    const {
-      formState: { isDirty, isValid },
-    } = methods;
+
+    const { handleSubmit, reset } = methods;
+
+    const [tabs, setTabs] = useState([
+      {
+        label: TabLabel.ACTIVE,
+        value: autoOpsRules.filter((rule) => !rule.triggeredAt),
+        selected: true,
+      },
+      {
+        label: TabLabel.COMPLETED,
+        value: autoOpsRules.filter((rule) => rule.triggeredAt),
+        selected: false,
+      },
+    ]);
+
+    useEffect(() => {
+      if (autoOpsRules?.length > 0) {
+        const ids = autoOpsRules
+          .filter((rule) => {
+            const { typeUrl } = rule.clausesList[0].clause;
+            const type = typeUrl.substring(typeUrl.lastIndexOf('/') + 1);
+            return type === ClauseType.EVENT_RATE && !rule.triggeredAt;
+          })
+          .map((rule) => rule.id);
+
+        if (ids.length > 0) {
+          dispatch(
+            listOpsCounts({
+              environmentNamespace: currentEnvironment.namespace,
+              ids,
+            })
+          );
+        }
+      }
+    }, [autoOpsRules]);
+
+    const handleClose = useCallback(() => {
+      reset();
+      history.replace({
+        pathname: `${PAGE_PATH_ROOT}${currentEnvironment.id}${PAGE_PATH_FEATURES}/${featureId}${PAGE_PATH_FEATURE_AUTOOPS}`,
```

Comment:
```suggestion
        pathname: `${PAGE_PATH_ROOT}${currentEnvironment.urlCode}${PAGE_PATH_FEATURES}/${featureId}${PAGE_PATH_FEATURE_AUTOOPS}`,
```

Reply:
@masaaania  There is no `urlCode` in `currentEnvironment`.

Reply:
Here is the type for currentEnvironment
<img width="290" alt="Screenshot 2023-09-28 at 12 58 05" src="https://github.com/bucketeer-io/bucketeer/assets/11039098/4b32f9be-7024-4700-8c84-918f77c5b45b">

Reply:
@bimalgrg519, please rebase your project from the main branch and regenerate the proto files again.

Reply:
@cre8ivejp  Thank you